webapp/src/main/java/com/google/sps/servlets/UploadServlet.java

```diff
@@ -76,13 +84,56 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
             // Save the file on disk
             item.write(storeFile);
 
-            // TO DO: Call validator here 
+            // TO DO: get feed name and thread specification from form
+            String validatorOutputJson = runValidator(filePath, 1, "nl-openov");
+
             response.getWriter().println("Upload has been done successfully!");
+            // Print the json output
+            response.getWriter().println(validatorOutputJson);
           }
         }
       }
     } catch (Exception ex) {
       response.getWriter().println("There was an error: " + ex.getMessage());
     }
   }
+
+  // Calls the global mobility validator function to load and validate the transit data
+  public static String runValidator(String filePath, int numThreads, String feedNameString) {
+    ValidatorLoader validatorLoader = new ValidatorLoader();
+    GtfsFeedLoader feedLoader = new GtfsFeedLoader();
+
+    GtfsFeedName feedName = GtfsFeedName.parseString(feedNameString);
+    final long startNanos = System.nanoTime();
+
+    // Input.
+    feedLoader.setNumThreads(numThreads);
+    NoticeContainer noticeContainer = new NoticeContainer();
+    GtfsFeedContainer feedContainer;
+    GtfsInput gtfsInput;
+
+    try {
+      gtfsInput = GtfsInput.createFromPath(Paths.get(filePath));
+    } catch (IOException e) {
+      e.printStackTrace();
+      return null;
+    }
+    feedContainer =
+        feedLoader.loadAndValidate(gtfsInput, feedName, validatorLoader, noticeContainer);
+
+    // Write the output to the temporary file: what happens when multiple files are written?
+    // Need a unique name ... or just write it to reponse and display it as HTML?
+    try {
+      Files.write(Paths.get("/tmp/", "report.json"),
+          noticeContainer.exportJson().getBytes(StandardCharsets.UTF_8));
+    } catch (IOException exception) {
+      exception.printStackTrace();
+    }
```

Comment:
Why do we need to write the report to a file?

Reply:
because the gtfs validator function reads the input from a file

Reply:
I am not referring to the input zip file. I am referring to the report file (report.json).

Reply:
Yes, that is what I was considering in the comment above. But I think you're right that we don't need to do this anymore